eZ/Publish/Core/Persistence/Legacy/Content/Location/Gateway/DoctrineDatabase.php

```diff
@@ -229,8 +229,12 @@ public function getSubtreeContent(int $sourceId, bool $onlyIds = false): array
         $statement = $query->execute();
 
         $results = $statement->fetchAll($onlyIds ? (FetchMode::COLUMN | PDO::FETCH_GROUP) : FetchMode::ASSOCIATIVE);
-        // array_map() is used to to map all elements stored as $results[$i][0] to $results[$i]
-        return $onlyIds ? array_map('reset', $results) : $results;
+        // array_map() is used to map all elements stored as $results[$i][0] to $results[$i]
+        return $onlyIds
+            ? array_map(static function ($result) {
```

Comment:
```suggestion
            ? array_map(static function (array $result): int {
```

but please double check return type declaration

Reply:
Return type depends on PDO variant. We should not assume `int` here.

As for the argument, :+1:

Reply:
Alternatively, we might add return type and casting to `int`. Your call @adamwojs. ;)

Reply:
After internal sync, we went only for specifying parameter type.